app/battle/battle.py

```diff
@@ -0,0 +1,22 @@
+from typing import Dict
+from app.knight import Knight
+
+
+class Battle:
+    def __init__(self) -> None:
+        pass
+
+    def fight(self, knight1: Knight, knight2: Knight) -> Dict[str, int]:
+        damage_to_knight1 = max(0, knight2.power - knight1.protection)
+        damage_to_knight2 = max(0, knight1.power - knight2.protection)
+
+        knight1.take_damage(damage_to_knight1)
+        knight2.take_damage(damage_to_knight2)
+
+        knight1.hp = max(knight1.hp, 0)
```

Comment:
You are directly modifying the `hp` attribute of the `Knight` class from outside of it. This is not a good practice as it breaks the encapsulation principle of OOP. The `hp` attribute should be private (i.e., `_hp`) and any modifications to it should be done through methods inside the `Knight` class. Consider creating a method in the `Knight` class that ensures the `hp` cannot fall below 0, and call this method instead of directly modifying the `hp` attribute here.